ohsome/clients.py

```diff
@@ -317,6 +329,15 @@ def _handle_request(self) -> OhsomeResponse:
             )
 
         except requests.exceptions.RequestException as e:
+            if isinstance(e, RetryError) and not self.__recursion_breaker:
+                # retry one last time without retries, this will raise the original error instead of a cryptic retry
+                # error (or succeed)
+                self._OhsomeBaseClient__session = None
+                self._OhsomeBaseClient__retry = False
+                # this should never happen because we set retry to False but just to be sure...
```

Comment:
If you are sure it will not happen please avoid adding redundant code. Is there a test that could confirm your statement?

Reply:
yes there is a test that enters this loop and asserts it only recurses once (as long as no unforeseen magic happens)

Reply:
see https://github.com/GIScience/ohsome-py/pull/128/files#diff-e13b5879c9cc60b9f41914c1a288d529677bb90637a29f59b9b4523473c59dc9R206